src/udev/scsi_id/scsi_id.c

```diff
@@ -58,12 +58,10 @@ static char revision_str[16];
 static char type_str[16];
 
 static void set_type(const char *from, char *to, size_t len) {
-        int type_num;
-        char *eptr;
+        unsigned type_num;
         const char *type = "generic";
 
-        type_num = strtoul(from, &eptr, 0);
-        if (eptr != from) {
+        if (safe_atou_full(from, 16, &type_num) >= 0) {
```

Comment:
you are changing the base from 0 to 16 here, that doesn't look right?

Reply:
https://github.com/systemd/systemd/pull/20013